api/src/test/java/io/opencensus/stats/StatsTest.java

```diff
@@ -74,8 +74,18 @@ public void setState_IgnoresInput() {
     assertThat(Stats.getState()).isEqualTo(StatsCollectionState.DISABLED);
   }
 
-  @Test(expected = NullPointerException.class)
+  @Test
   public void setState_DisallowsNull() {
+    thrown.expect(NullPointerException.class);
+    thrown.expectMessage("state");
     Stats.setState(null);
   }
+
+  @Test
+  public void disallowSetStateAfterGetState() {
```

Comment:
It looks like this test could cause other tests to fail if the tests ran in the right order, since Stats has global state.  It would probably be better to only test this functionality in NoopStatsTest.  Another option is to combine all of the state testing into one method and add a comment.

Reply:
Oh that's true, I've removed this test since this should have been covered in NoopStatsTest.